library/Reporting/Scheduler.php

```diff
@@ -149,27 +149,23 @@ protected function add(Schedule $schedule)
         }
 
         $next = clone $start;
-        $next->modify($modify);
-        $interval = $next->getTimestamp() - $start->getTimestamp();
-
-        $current = $start->getTimestamp() - $now->getTimestamp();
 
         printf("Scheduling job %s to run at %s.\n", $name, $start->format('Y-m-d H:i:s'));
 
-        $loop = function () use (&$loop, $name, $callback, $interval, $schedule) {
+        $loop = function () use (&$loop, $name, $callback, $next, $modify, $schedule) {
             $callback();
 
-            $nextRun = (new \DateTime())
-                ->add(new \DateInterval("PT{$interval}S"));
+            $startTimestamp = $next->getTimestamp();
```

Comment:
This should be the current time, otherwise the timer is, once delayed, always delayed.

- $now
- 1 month after
- 2 months and 5 minutes after (+5 minutes delay)
- 3 months and 5 minutes after
- 4 months and 10 minutes after (+5 minutes delay)
- 5 months and 10 minutes after

With months this isn't that prevalent, but the shorter the interval the more it is.